x/gov/deposit.go

```diff
@@ -0,0 +1,122 @@
+	// Checks to see if proposal exists
+	proposal, ok := keeper.GetProposal(ctx, proposalID)
+	if !ok {
+		return ErrUnknownProposal(keeper.codespace, proposalID), false
+	}
+
+	// Check if proposal is still depositable
+	if (proposal.Status != StatusDepositPeriod) && (proposal.Status != StatusVotingPeriod) {
+		return ErrAlreadyFinishedProposal(keeper.codespace, proposalID), false
+	}
+
+	// Send coins from depositor's account to DepositedCoinsAccAddr account
+	// TODO: Don't use an account for this purpose; it's clumsy and prone to misuse.
+	err := keeper.ck.SendCoins(ctx, depositorAddr, DepositedCoinsAccAddr, depositAmount)
+	if err != nil {
+		return err, false
+	}
+
+	// Update proposal
+	proposal.TotalDeposit = proposal.TotalDeposit.Add(depositAmount)
+	keeper.SetProposal(ctx, proposal)
+
+	// Check if deposit has provided sufficient total funds to transition the proposal into the voting period
+	activatedVotingPeriod := false
+	if proposal.Status == StatusDepositPeriod && proposal.TotalDeposit.IsAllGTE(keeper.GetDepositParams(ctx).MinDeposit) {
+		keeper.activateVotingPeriod(ctx, proposal)
+		activatedVotingPeriod = true
+	}
+
+	// Add or update deposit object
+	currDeposit, found := keeper.GetDeposit(ctx, proposalID, depositorAddr)
+	if !found {
+		newDeposit := NewDeposit(proposalID, depositorAddr, depositAmount)
+		keeper.setDeposit(ctx, proposalID, depositorAddr, newDeposit)
+	} else {
+		currDeposit.Amount = currDeposit.Amount.Add(depositAmount)
+		keeper.setDeposit(ctx, proposalID, depositorAddr, currDeposit)
+	}
+
+	return nil, activatedVotingPeriod
+}
+
+// GetAllDeposits returns all the deposits from the store
+func (keeper Keeper) GetAllDeposits(ctx sdk.Context) (deposits Deposits) {
+	keeper.IterateAllDeposits(ctx, func(deposit Deposit) bool {
+		deposits = append(deposits, deposit)
+		return false
+	})
+	return
+}
+
+// GetDeposits returns all the deposits from a proposal
+func (keeper Keeper) GetDeposits(ctx sdk.Context, proposalID uint64) (deposits Deposits) {
+	keeper.IterateDeposits(ctx, proposalID, func(deposit Deposit) bool {
+		deposits = append(deposits, deposit)
+		return false
+	})
+	return
+}
+
+// GetDepositsIterator gets all the deposits on a specific proposal as an sdk.Iterator
+func (keeper Keeper) GetDepositsIterator(ctx sdk.Context, proposalID uint64) sdk.Iterator {
+	store := ctx.KVStore(keeper.storeKey)
+	return sdk.KVStorePrefixIterator(store, types.DepositsKey(proposalID))
+}
+
+// RefundDeposits refunds and deletes all the deposits on a specific proposal
+func (keeper Keeper) RefundDeposits(ctx sdk.Context, proposalID uint64) {
+	store := ctx.KVStore(keeper.storeKey)
+
+	keeper.IterateDeposits(ctx, proposalID, func(deposit Deposit) bool {
+		err := keeper.ck.SendCoins(ctx, DepositedCoinsAccAddr, deposit.Depositor, deposit.Amount)
+		if err != nil {
+			panic("should not happen")
```

Comment:
```suggestion
			panic(fmt.Sprintf("Refund deposits can't send %s from %s to %s", deposit.Amount, DepositedCoinsAccAddr, deposit.Depositor))
```

Reply:
what about `panic(err)` ?

Reply:
still valid, I was thinking some context may help but the original error may be cleaner